neo/SmartContract/WildCardContainer.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Newtonsoft.Json;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Neo.SmartContract
+{
+    public class WildCardContainer<T> : IReadOnlyList<T>
+    {
+        private readonly T[] _data;
+
+        [JsonIgnore]
+        public T this[int index] => _data[index];
+
+        /// <summary>
+        /// Number of items
+        /// </summary>
+        [JsonIgnore]
+        public int Count => _data?.Length ?? 0;
+
+        /// <summary>
+        /// Is will card?
+        /// </summary>
+        [JsonIgnore]
+        public bool IsWildcard => _data != null ? _data.Length == 0 : true;
```

Comment:
```suggestion
        public bool IsWildcard => _data is null;
```

`_data.Length == 0` doesn't mean wildcard. For example:

```json
{
  "permissions": []
}
```

This means it doesn't need any permission, not needs "*" permissions.